### DISCUSSION TITLE
Support for Hub and Spoke Connectivity?

### DISCUSSION THREAD

**Discussion by awm-achi** [skupperproject/skupper#891] [Q&A]
We have a central management cluster, and edge clusters. We would like to connect the edge clusters to the central cluster, without exposing the edge cluster services to other edge cluster services but I can't seem to figure out how to get it to work. Research online says disabling service-sync should work, but  that causes all services to be disabled, directly exposed services included. Is there any guidance or documentation about how service-sync works, or how I might configure my connectivity to work the way I want it to?

**Comment by grs** [comment]
There are two approaches I can suggest. One is to have a single skupper network extending to all the edges, but disable service-sync and manually create the services for all the edges in the management cluster (with `skupper service create`).

The second would be to have a separate 'network' linking each edge to the central cluster. This would require a distinct namespace to be used for each edge on the central cluster, through which the services for that edge would be accessed.

**Reply by awm-achi** [reply]
for the first approach, I did as described except I used skupper expose service rather than skupper service create. What is the difference between the two and is there documentation on skupper service create? I was not able to get skupper expose working when service-sync was disabled on the central cluster.

**Reply by grs** [reply]
`skupper service create` defines a skupper service in the namespace it is invoked in. However defining the service is not enough, somewhere in the network you need to also do a `skupper service bind` to bind an implementation to which service requests can be routed. `skupper expose` is a combination of `skupper service create` and `skupper service bind`

So you can do `skupper expose` in the edge cluster, but then only `skupper service create` is needed in the central cluster for that edge service. (Service sync is responsible for automatically propagatig service definitions, meaning that if enabled, you only need the service to be defined once in the whole network).

**Reply by awm-achi** [reply]
Ok, I think I understand, so I can use Skupper expose service xxxx --port yyyy --address zzzzz on the edge cluster and just do Skupper service create zzzzz yyyy on the main cluster? And is there a way to do multiple ports? Or is it just Skupper service create zzzzz aaaa again?

**Reply by grs** [reply]
> Ok, I think I understand, so I can use Skupper expose service --port yyyy --address on the edge cluster and just do Skupper service create on the main cluster? 

Yes, that is correct. 

> And is there a way to do multiple ports? Or is it just Skupper service create again?

You can add multiple ports after the service name, e.g. ```skupper service create foo 8080 8081 8082```

**Reply by awm-achi** [reply]
Ok and if I need to share a service from the hub down to the edge cluster is it the same process? Service expose on the central cluster and service create on the edge?

**Reply by grs** [reply]
correct

**Reply by awm-achi** [reply]
Ok this worked! Will there ever be a feature toggle to block indirect connections? Since it seems like Skupper knows which connections are indirect it seems like something that might be useful.

**Reply by grs** [reply]
Certainly we recognise the need for more control over the exposure of services. I doubt it would be in the form of allowing or disallowing indirect connections, but this is the first time I've heard that requested so I will think on it some more.
